hearthstone/simulator/core/card_pool.py

```diff
@@ -2624,6 +2625,20 @@ def handle_event_powers(self, event: CardEvent, context: Union[BuyPhaseContext,
                         context.owner.play_blood_gem(card)
 
 
+class ArchdruidHamuul(MonsterCard):
+    tier = 6
+    base_attack = 4
+    base_health = 4
+    monster_type = MONSTER_TYPES.NEUTRAL
+    legendary = True
+
+    def base_battlecry(self, targets: List[MonsterCard], context: BuyPhaseContext):
+        most_common_types = set(card.monster_type for card in context.owner.in_play if card.monster_type in MONSTER_TYPES.single_types())
+        if most_common_types:
+            most_common_type = most_common_types.pop()
```

Comment:
I don't think this will give the most common type, just a random types. (But also not using the randomizer, so it will break the determinism of the game!).

Reply:
oops!